### PR TITLE
add sortByValue argument to BarChart

### PR DESCRIPTION
**Pull request by unclejustin**
Closes https://github.com/sourcegraph/sourcegraph/issues/41256

| Before | After |
| --- | --- |
|PixelSnap 2022-09-07 at 11 55 06|PixelSnap 2022-09-07 at 11 55 50|

## Test plan

Run this query in search `python2\.([0-9]) repo:github\.com/sourcegraph`
Should render aggregation graph ordered by value descending.

## App preview:

- [Web](https://sg-web-insights-sort-chart-by-value.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-gbaalhbola.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/wildcard/src/components/Charts/components/bar-chart/BarChart.story.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ interface LanguageUsageDatum {
 const LANGUAGE_USAGE_DATA: LanguageUsageDatum[] = [
     {
         name: 'JavaScript',
-        value: 422,
+        value: 129,
```
Comment (unclejustin) [inline client/wildcard/src/components/Charts/components/bar-chart/BarChart.story.tsx:38]: Note: I changed these values to make it more obvious how the sorting changes.

File: client/wildcard/src/components/Charts/components/bar-chart/utils/get-grouped-categories.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { groupBy } from 'lodash'
 interface GetCategoriesInput<Datum> {
     data: Datum[]
     stacked: boolean
+    sortByValue?: boolean
```
Comment (unclejustin) [inline client/wildcard/src/components/Charts/components/bar-chart/utils/get-grouped-categories.ts:6]: To keep this PR simple, I made this an optional boolean. We should probably consider an enum in the future or pass a "sortBy" function.

Comment (unclejustin) [reply client/wildcard/src/components/Charts/components/bar-chart/utils/get-grouped-categories.ts:6]: I experimented with passing a `sortBy` function, but it requires that you know about the bar charts grouping internals. That may be ok, but we would have to expose the `Category` type to do so.

**Review by vovakulikov — APPROVED** [review]
LGTM. @unclejustin let file an issue about tech debt around this newly created props only for the aggregation UI or we can leave a comment to this issue about low level API https://github.com/sourcegraph/sourcegraph/issues/40009

**Pull request merged**